test/unit/nedap/speced/def/spec_assertion.cljc

```diff
@@ -53,3 +54,23 @@
   (is (spec-assertion-thrown? 'number? (returns-number-fn "1234")))
   (is (spec-assertion-thrown? 'string? (accepts-string (->ExampleRecord) 1234)))
   (is (spec-assertion-thrown? 'string? (returns-string (->ExampleRecord) 1234))))
+
+(defn parameterized-defn [spec x]
+  (check! spec x)
+  x)
+
+#?(:clj (defmacro when-not-u-s-1-0-0
+          {:style/indent 0}
+          [& body]
+          (when-not (-> "nedap.speced.def.testing.utils-spec-dep" System/getProperty #{"1.0.0"})
```

Comment:
So far we're not putting arbitrarily many dependencies into this setup; but just one - namely the "latest u.spec dependency prior to introducing https://github.com/nedap/speced.def/pull/99".

Since there's just one version to watch out for, I'd keep things fixed and rather predictable (a dynamic system is always more indirect).

If at some point we needed to actively test _more than two_ u.s versions in or build, we could revise this solution. Undoubtedly that solution would be subject to code review and a qa plan so there's not much room for error

Reply:
> Since there's just one version to watch out for,

Is that the case? wouldn't this test fail with utils.spec 1.0.0 and 1.1.0 ?

> I'd keep things fixed and therefore predictable

This makes sense 👍

Reply:
> Is that the case? wouldn't this test fail with utils.spec 1.0.0 and 1.1.0 ?

1.0.0 and 1.1.0 are equivalent, for the aspect we are testing here.

So it doesn't matter which one we pick, and (conceptually) both are covered by the updated test suite.

Answering your literal question: yes, those deps would fail against the _new_ test, which is the precise reason why there's `(when-not (-> "nedap.speced.def.testing.utils-spec-dep" System/getProperty #{"1.1.0"})`